src/mod_stream_management.erl

```diff
@@ -128,17 +293,74 @@ register_smid(SMID, SID) ->
               {error, Reason}
     end.
 
-get_sid(SMID) ->
-    case mnesia:dirty_read(sm_session, SMID) of
-        [#sm_session{sid = SID}] ->
-            [SID];
-        [] ->
-            []
+register_stale_smid_h(SMID, H) ->
+    case gen_mod:get_module_opt(?MYNAME, ?MODULE, stale_h, {false, []}) of
+        {false, []} -> ok;
+        {true, _} ->
+            try
+                Stamp = now_to_seconds(erlang:timestamp()),
+                mnesia:sync_dirty(fun mnesia:write/1,
```

Comment:
How about just using:
http://erlang.org/doc/man/mnesia.html#dirty_write-1
to simplify the code a bit?